src/nc_ssl.c

```diff
@@ -290,6 +290,50 @@ nc_ssl_read(SSL *ssl, void *buf, size_t num) {
         }
     }
 
+    log_debug(LOG_VERB, "read %d initial bytes", bytes_read);
+
+    // OpenSSL reads data in record size blocks (16kb for SSLv3/TLSv1, see
+    // SSL_read man page for details). Sometimes, SSL_read returns data but
+    // there is still more data available from the next block. Keep reading
+    // until there is actually no more data.
+    //
+    // Don't use SSL_pending since it does not reliably work.
+    for (;;) {
+        void *next_buf = buf + bytes_read;
+        void *bytes_remaining = num - bytes_read;
+        int ssl_read_status = SSL_read(ssl, next_buf, (int)bytes_remaining);
```

Comment:
What happens if bytes_remaining is 0?

Reply:
Good question. I think `SSL_read` will handle it, but would prefer to be explicit.

Also bytes_remaining shouldn't be a `void*`... will fix that.